src/Vault.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "lib/chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+
+
+interface IERC20Decimals is IERC20 {
+    function decimals() external view returns (uint);
+}
+
+contract Vault is Ownable {
+
+    ///@dev Vault records collateral deposits of each user
+    mapping (address => uint) public deposits; 
+
+    ///@dev Vault records debt holdings of each user
+    mapping (address => uint) public debts;  
+
+    ///@dev ERC20 interface specifying token contract functions
+    IERC20Decimals public immutable collateral;    
+    IERC20Decimals public immutable debt;  
+
+    ///@dev Asset Pricefeed interface from Chainlink
+    AggregatorV3Interface public immutable priceFeed;   
+    
+    ///@dev Emitted on deposit()
+    ///@param collateralAsset The address of the collateral asset
+    ///@param user The address of the user calling deposit()
+    ///@param collateralAmount The amount of collateral asset deposited
+    event Deposit(address indexed collateralAsset, address indexed user, uint collateralAmount);  
```

Comment:
Thought it would be a nice touch to allow for the possibility of this scaling into multi-collateral. Didn't consider if the gas effects would be significant though.

Would you prefer if I leave it as is, or modify the code and tests to drop the extra topics entirely?

Reply:
using the words "allow for the possibility" in an explanation is usually a code smell that ur overengineering smthg

if this were a yield pr we would def take it out

Reply:
Gotcha, will remove it